app/shop.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+import datetime
+from typing import List
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list,
+            products: dict
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def calculate_amount_for_products(self, product_cart: dict) -> int | float:
+        total_amount = 0
+        for product_name, count in product_cart.items():
+            total_amount += self.products.get(product_name, 0) * count
+        return total_amount

```

Comment:
try to use `sum()` and  comprehension

Reply:
Thx a lot ! 
`sum()` and comprehension is a much better solution.